src/Process/LayeredMaterialNodeProcessing.js

```diff
@@ -228,11 +228,19 @@ export function updateLayeredMaterialNodeImagery(context, layer, node) {
     }
 
     const currentLevel = node.material.getColorLayerLevelById(layer.id);
-    const zoom = node.getCoordsForLayer(layer)[0].zoom || node.level;
-    const targetLevel = chooseNextLevelToFetch(layer.updateStrategy.type, node, zoom, currentLevel, layer);
+    const bestLevel = node.getCoordsForLayer(layer)[0].zoom || node.level;
+    let targetLevel = node.layerUpdateState[layer.id].targetLevel ? node.layerUpdateState[layer.id].targetLevel - 1 :
+        chooseNextLevelToFetch(layer.updateStrategy.type, node, bestLevel, currentLevel, layer);
+
+    targetLevel = Math.min(layer.options.zoom.max, targetLevel);
+
     if (targetLevel <= currentLevel) {
         return;
     }
+    if (!layer.tileInsideLimit(node, layer, targetLevel)) {
```

Comment:
This has already been tested above. Could you add a comment explaining why this 2nd check is needed?

Reply:
Because if the first test `layer.tileInsideLimit` returns that it is out of limits and the node inherits from its parent, then it'll still make a command to fetch texture.
and you must check with the `targetLevel`

Reply:
This comment should be in the code; otherwise someone may remove this thinking it's useless.

And I believe this shouldn't be done here. But rather around l.156 where we explicitely ignore the `tileInsideLimit` result.

Reply:
it's more difficult to do it higher because we have to consider the `targetLevel`

Reply:
Does it really depends on `targetLevel`?

Couldn't be fixed by always calling `node.layerUpdateState[layer.id].noMoreUpdatePossible();` here:
```js
if (!layer.tileInsideLimit(node, layer)) {
 node.layerUpdateState[layer.id].noMoreUpdatePossible();
 ...
}
```

This would allow to inherit the parent's texture where possible, but would prevent any further fetch. WDYT?

Reply:
if the parent has a down-scaled texture, inheritance'll not provide the best texture

by example a layer with limit level 0 to 10, a node with level 12  can inherit a texture with level 0.
and have in this case its status to `UPDATE_STATE.FINISHED` and then never get the level 10 texture

Reply:
OK I get it.

The problem here is that `tileInsideLimit` can be expensive - especially for StaticProvider which uses a naive search algo. Anyway your change makes sense and StaticProvider can be improved.